src/parser.rs

```diff
@@ -1683,6 +1689,19 @@ impl Parser {
         })
     }
 
+    fn parse_show_objects(&mut self) -> Option<Statement> {
+        let object_type = self
+            .parse_one_of_keywords(&["SOURCES", "VIEWS", "SINKS", "TABLES"])
+            .map(|s| match s {
```

Comment:
I think if this was structured as `match self.parse_one_of_keywords(..)` instead of mapping over the parse result you'll have an easier time returning `Ok`/`Err`.